atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+                return insndesc['content']  # we will reuse the line verbatim
+            if insndesc['instruction'] == 'FROM':
+                break  # no USER specified in final stage
+
+    def _cleanup_lines(self):
+        lines = [
+            "RUN rm -f " + " ".join(
+                (f"'{repo.dst_filename}'" for repo in self.yum_repos)
+            )
+        ]
+        if self._builder_ca_bundle:
+            lines.append(f'RUN rm -f /tmp/{self._ca_bundle_pem}')
+
+        final_user_line = self._final_user_line()
+        if final_user_line:
+            lines.insert(0, "USER root")
+            lines.append(final_user_line)
+
+        return lines
+
+    def add_koji_repo(self):
+        self.xmlrpc = get_koji_session(self.workflow.conf)
+        self.pathinfo = self.workflow.conf.koji_path_info
+        self.proxy = self.workflow.conf.yum_proxy
+
+        if not self.target:
+            self.log.info('no target provided, skipping plugin')
+            return
+
+        target_info = self.xmlrpc.getBuildTarget(self.target)
+        if target_info is None:
+            self.log.error("provided target '%s' doesn't exist", self.target)
+            raise RuntimeError("Provided target '%s' doesn't exist!" % self.target)
+        tag_info = self.xmlrpc.getTag(target_info['build_tag_name'])
+
+        if not tag_info or 'name' not in tag_info:
+            self.log.warning("No tag info was retrieved")
+            return
+
+        repo_info = self.xmlrpc.getRepo(tag_info['id'])
+
+        if not repo_info or 'id' not in repo_info:
+            self.log.warning("No repo info was retrieved")
+            return
+
+        # to use urljoin, we would have to append '/', so let's append everything
+        baseurl = self.pathinfo.repo(repo_info['id'], tag_info['name']) + "/$basearch"
+
+        self.log.info("baseurl = '%s'", baseurl)
+
+        repo = {
+            'name': 'atomic-reactor-koji-plugin-%s' % self.target,
+            'baseurl': baseurl,
+            'enabled': 1,
+            'gpgcheck': 0,
+        }
+
+        # yum doesn't accept a certificate path in sslcacert - it requires a db with added cert
+        # dnf ignores that option completely
+        # we have to fall back to sslverify=0 everytime we get https repo from brew so we'll surely
+        # be able to pull from it
+
+        if baseurl.startswith("https://"):
+            self.log.info("Ignoring certificates in the repo")
+            repo['sslverify'] = 0
+
+        if self.proxy:
+            self.log.info("Setting yum proxy to %s", self.proxy)
+            repo['proxy'] = self.proxy
+
+        yum_repo = YumRepo(os.path.join(YUM_REPOS_DIR, self.target))
+        path = yum_repo.dst_filename
+        self.log.info("yum repo of koji target: '%s'", path)
+        yum_repo.content = render_yum_repo(repo, escape_dollars=False)
+        self.yum_repos.append(yum_repo)
+
+    def __init__(self, workflow, target=None, repourls=None, inject_proxy=None):
```

Comment:
I didn't expect a constructor this far down :smile: 

It's just cosmetics, but could this be moved above the other methods?

Reply:
moved up